### PR TITLE
Update block attributes for clarification

### PR DESCRIPTION
**Pull request by mkaz**
## Description

Edit introduction to attributes for clarification.

Move up the object keys section since it was lost in the hpq paragraph, also the keys are specified first, so better to mention them first.

Add example for using class as a selector

## Types of changes

Documentation. [View live on branch](https://github.com/WordPress/gutenberg/blob/docs/attrib-clarify/docs/designers-developers/developers/block-api/block-attributes.md)

### CODE REVIEW

File: docs/designers-developers/developers/block-api/block-attributes.md [hunk 1]
```diff
@@ -2,13 +2,18 @@
 
 ## Common Sources
 
-Attribute sources are used to define the strategy by which block attribute values are extracted from saved post content. They provide a mechanism to map from the saved markup to a JavaScript representation of a block.
+Attribute sources are used to define how the block attribute values are extracted from saved post content. They provide a mechanism to map from the saved markup to a JavaScript representation of a block.
 
 If no attribute source is specified, the attribute will be saved to (and read from) the block's [comment delimiter](/docs/designers-developers/key-concepts.md#delimiters-and-parsing-expression-grammar).
 
-Each source accepts an optional selector as the first argument. If a selector is specified, the source behavior will be run against the corresponding element(s) contained within the block. Otherwise it will be run against the block's root node.
+The keys specified in the attributes source object are named as you see fit. The result of the attribute source definition will be assigned as a value to each key.
```
Comment (jorgefilipecosta) [inline docs/designers-developers/developers/block-api/block-attributes.md]: Should we to "The result of the attribute source definition will be assigned as a value to each key." to "The result of the attribute source definition is assigned as a value to each key." as it looks even more direct and in accordance with the rest of the text e.g: the source definition runs against ...

File: docs/designers-developers/developers/block-api/block-attributes.md [hunk 2]
```diff
@@ -2,13 +2,18 @@
...
+
+If no selector argument is specified, the source definition runs against the block's root node. If a selector argument is specified, it will run against the specified element(s) contained within the block.
+
+The selector specified can be an HTML tag, but also anything queryable such as a class or id attribute, see examples below.
```
Comment (jorgefilipecosta) [inline docs/designers-developers/developers/block-api/block-attributes.md]: Maybe but also -> or?

File: docs/designers-developers/developers/block-api/block-attributes.md [hunk 3]
```diff
@@ -2,13 +2,18 @@
...
+
+Under the hood, attribute sources are a superset of functionality provided by [hpq](https://github.com/aduth/hpq), a small library used to parse and query HTML markup into an object shape.
```
Comment (jorgefilipecosta) [inline docs/designers-developers/developers/block-api/block-attributes.md]: Maybe "superset of functionality provided" to "superset of the functionality provided"?

**Review by jorgefilipecosta — APPROVED** [review]
The changes look good to me and I like this more direct writing style 👍  I left some suggestions as possible enhancements but I am not too certain about them so feel free to ignore the suggestions.

File: docs/designers-developers/developers/block-api/block-attributes.md (hunk 1 above)
Comment (mkaz) [reply docs/designers-developers/developers/block-api/block-attributes.md]: :+1:

File: docs/designers-developers/developers/block-api/block-attributes.md (hunk 2 above)
Comment (mkaz) [reply docs/designers-developers/developers/block-api/block-attributes.md]: :+1:

File: docs/designers-developers/developers/block-api/block-attributes.md (hunk 3 above)
Comment (mkaz) [reply docs/designers-developers/developers/block-api/block-attributes.md]: :+1:  I like all the suggestions, thanks!

**Pull request merged**